### DISCUSSION TITLE
Fragen zur Konfiguration

### DISCUSSION THREAD

**Discussion by tomltom** [hoylabs/OpenDTU-OnBattery#1220] [Q&A]
Hallo,
Nach dem ich mich so langsam durch alle Tücken der Hardware gekämpft habe, 
Hänge ich jetzt ein wenig an der Konfiguration.
 
Manche Schalter und Einstellungen erschließen sich einem erst nach der Nutzung.
Oder man hat die Software nach seinen eigenen Vorstellungen programmiert.

Besten Dank mal hier an alle Beteiligten   !!!!!!!
 
Meine Hardware Konfiguration:

ESP32S3 16MB (SPI Flash)
Huawei R4850G2 per CAN Bus MCP2515
Das Slotdetect per 1 Kanal Optokoppler Relais
Pylontech US 3000C per CAN Bus WaveShare SN65HVD230
HM-800 per nRF24+ 
Display SSD1306
(falls jemand die pin_config haben möchte….  PN an mich)
 
SW Release 2024.08.18

Diese Release habe ich zuerst nicht zum laufen bekommen.
Jedesmal wenn der Huawei CAN dazu kam (und zwar nur mit angeschlossener HW)
lief der ESP in einen Boot Loop.
Nach einem Google Marathon … 
es gab einen Fix der den CAN Stack geändert hat.
Commit 119bd3a
 
Jetzt habe ich  mir auch noch den Marathon mit VS Code und diversen Bibliotheken angetan
Und mir eine eigene FW kompiliert .
Und es Läuft!!!
 
So, jetzt zum eigentlichen Problem.
Alles unterhält sich, aber nix wird gemacht.
Hier ist die Stelle, die mir merkwürdig vorkommt
 
17:00:54.344 > [DPL::loop] ******************* ENTER **********************
17:00:54.397 > [DPL::loop] battery interface enabled, SoC: 50 %, StartTH: 80 %, StopTH: 20 %, SoC age: 1 s, ignore: no
17:00:54.443 > [DPL::getBatteryVoltage] BMS: 49.37 V, MPPT: -1.00 V, inverter: 49.40 V, returning: 49.37V
17:00:54.494 > [DPL::loop] dcVoltage: 49.37 V, loadCorrectedVoltage: 49.37 V, StartTH: 50.00 V, StopTH: 49.00 V
17:00:54.543 > [DPL::loop] StartTH reached: no, StopTH reached: no, SolarPT enabled, use at night: no
17:00:54.592 > [DPL::calcPowerLimit] battery use prevented, solar power (DC): 0 W
**17:00:54.637 > [DPL::announceStatus] no energy source available to power the inverter from**
17:00:54.757 > [DPL::announceStatus] the system is stable, the last power limit is still valid
 
Wieso hat der Inverter keine Energie Quelle?

Mein eigentliches Ziel ist, bei ausreichend Solar die Batterie(n) zu laden,
und am Abend den gespeicherten Überschuß wieder zu verwenden.
Null Einspeisung ist nicht nötig.

Kann mir bitte jemand ein paar Tips zur Konfiguration geben??
Falls noch Angaben notwendig sind, kann ich liefern.
Wollte hier nicht unnötig alles voll laden.

Besten Dank
Tom

**Comment by spcqike** [comment]
> 17:00:54.494 > [DPL::loop] dcVoltage: 49.37 V, loadCorrectedVoltage: 49.37 V, StartTH: 50.00 V, StopTH: 49.00 V

die Batterie ist noch nicht ausreichend geladen.

**Reply by tomltom** [reply]
Oh.. schon mal ein guter Hinweis. Danke

Aber wie sag ich der Regelung jetzt ...bitte laden !
Ich habe dem Gerät "Stromzähler" per MQTT erzählt, das 500 W über Solar da sind.
Es wurde auch bei Leistung/Netzt -500W angezeigt. (Minus ist doch in diesem Fall korrekt ?? )
Aber dem Ladegerät (Huawei) wurde nicht mitgeteilt, dreh mal auf .

Ich habe beim AC Ladegerät die Option "Automatische Leistungssteuerung" eigeschaltet. Alles Andere auf Standard Werte:
Das selbe beim "Dynamic Power Limiter", eingeschaltet und den Rest auf Standard gelassen.

Ich habe leider keine zusammenhängende Dokumentaion zum zusammenspiel aller Optionen gefunden.

Danke
Tom

**Reply by spcqike** [reply]
Das kann ich dir leider auch nicht sagen, ich verwende kein Ladegerät.

>  Es wurde auch bei Leistung/Netzt -500W angezeigt. (Minus ist doch in diesem Fall korrekt ?? )

ja.

> Ich habe beim AC Ladegerät die Option "Automatische Leistungssteuerung" eigeschaltet. Alles Andere auf Standard Werte:
Das selbe beim "Dynamic Power Limiter", eingeschaltet und den Rest auf Standard gelassen.

zeig doch einfach mal deine Einstellungen, die du nun hast. dann kommt man vielleicht weiter 😄 

mehr als die Beschreibung aus dem Wiki kann ich dir auch nicht dazu geben
https://github.com/helgeerbe/OpenDTU-OnBattery/wiki/Huawei-AC-PSU#operation-modes

@schlimmchen 
kann es sein, dass der AC Charger in der neuen Dokumentation noch komplett fehlt? :)

**Reply by tomltom** [reply]
Jetzt hab ich es verstanden.
Das Ladegerät lädt nicht weil meine "Start Spannungslimit: " kleiner der aktuellen Batteriespannung ist.
Erst wenn die Batterie Spannung unter diesen Wert fällt , 
wird das Ladegerät mit dem zur Verfügung stehendem Überschuss angesteuert bis das Ladespannungslimit erreicht ist.
Oder zuwenig Leistung vorhanden ist.

Vielen Dank für Hilfe, am Ende war es wieder mal nur RTMF.
Danke!
Tom

**Reply by tomltom** [reply]
Und Ja, der AC Charger ist noch nicht in der neuen Dokumentation .

**Reply by schlimmchen** [reply]
> kann es sein, dass der AC Charger in der neuen Dokumentation noch komplett fehlt? :)

Leider ist das richtig. Das liegt daran, dass ich davon keinen Plan habe und die Infos im Wiki mir zu wild waren um sie lediglich zu kopieren. Hilfe ist willkommen.

**Reply by spcqike** [reply]
Ah ok. Kann ich nachvollziehen. 

Leider kann ich auch nichts dazu beitragen. Ist mir nur aufgefallen und dachte, vielleicht ist es in der Vielzahl an Seiten einfach unter gegangen (:

**Reply by tomltom** [reply]
So, jetzt redet ds System und die Batterie wird auch geladen.
Allerdings kommen mir die Werte komisch vor
![dtu_live](https://github.com/user-attachments/assets/9d362458-7469-43fc-95ce-080fccf4a4c3)

Ich sage der DTU per MQTT ich  habe 300W übrig, aber der Huawei lädt  mit ca. 1000W die Batterie.
Der Log sagt dazu 
12:45:25.577 > [DPL::loop] ******************* ENTER **********************
12:45:25.577 > [DPL::loop] battery interface enabled, SoC: 55 %, StartTH: 80 %, StopTH: 20 %, SoC age: 0 s, ignore: no
12:45:25.577 > [DPL::getBatteryVoltage] BMS: 50.72 V, MPPT: -1.00 V, inverter: 51.00 V, returning: 50.72V
12:45:25.577 > [DPL::loop] dcVoltage: 50.72 V, loadCorrectedVoltage: 50.72 V, StartTH: 50.00 V, StopTH: 49.00 V
12:45:25.577 > [DPL::loop] StartTH reached: no, StopTH reached: no, SolarPT enabled, use at night: no
12:45:25.577 > [DPL::calcPowerLimit] battery use prevented, solar power (DC): 0 W
12:45:25.577 > [DPL::announceStatus] no energy source available to power the inverter from
12:45:25.577 > [DPL::announceStatus] the system is stable, the last power limit is still valid

Wo steckt mein Fehler?

Wenn ich mal alles am Laufen habe, kann ich gerne was zur Doku beisteuern.

**Reply by spcqike** [reply]
> Wo steckt mein Fehler?

> Ich habe dem Gerät "Stromzähler" per MQTT erzählt, das 500 W über Solar da sind.

reduzierst du dein (simulierten?) Stromzählerwert, wenn der Huawei lädt?

Wenn du -300W hast, und das System 300W lädt, 0W am Zähler erwartet, aber weiterhin -300W übrig sind, warum sollte es nicht "aufdrehen"?

**Reply by tomltom** [reply]
So ist, ich Trottel.
Und klar, maximale Leistung AC Ladegerät sind auf 1000W gestellt.
DTOB geht jetzt in den Praxis Betrieb, und wird die nächsten Tage beobachtet.

Besten Dank an alle
